### DISCUSSION TITLE
✅ Windows and copy paste

### DISCUSSION THREAD

**Discussion by photochap** [isontheline/pro.webssh.net#455] [Ideas]
Hello
1- is it possible to have multi-window view, see photo?
![Capture d’écran 2021-11-15 à 18 10 52](https://user-images.githubusercontent.com/6159635/141824670-6586f1b4-4dc6-4cb8-aa01-2a8d8b11c95c.png)

2- is it possible to copy paste easier ? 
3- when windows changing the focus is deactivate, i must click on windows before write something. can you program focus on window when is selected ?
4- send pre-registered shell commands 
5- it is not easy to navigate in terminal history. Exemple if i do enter or clear the cursor do not return on the command line.
Thanks

**Comment by isontheline** [comment]
Bonjour Christophe 👋

1. La seule possibilité, pour le moment, de faire du multi-terminal est celle ci : https://webssh.net/documentation/help/howtos/launching-multiple-terminals/
2. OK j'ouvre une issue pour pouvoir faire Cmd-C Cmd-V à la place du clic prolongé pour le copier / coller : https://github.com/isontheline/pro.webssh.net/issues/456
3. OK j'ouvre une issue pour que le focus revienne lors du changement de fenêtre : https://github.com/isontheline/pro.webssh.net/issues/457
4. Les macros ? Elles se trouvent dans la troisième icône à droite du terminal : </>
5. Pouvez-vous poster un exemple photo / vidéo du problème rencontré ?

Bonne soirée à vous

**Reply by photochap** [reply]
Bonjour 
Merci pour tous vos retour
5 - Lorsque avec la sourie vous remontez dans le terminale (Buffer du terminal) et que sur la ligne de commande vous tapez la cmd >clear. la command s'excuse mais la ligne de cmd ne revient pas haut. si vous aver un terminal apple après avoir consulter le buffer du terminal et tapper la cmd clear, la ligne de cmd revient en haut du terminal
![Buffer_du_terminal](https://user-images.githubusercontent.com/6159635/142209776-e93c5046-a1b5-4424-8655-985e4de6b59b.png)
e
![cmdClear](https://user-images.githubusercontent.com/6159635/142209791-38fee2ca-412a-4090-8b68-9d1230c339af.png)
 
Bonne continuation
Cdlt

**Reply by isontheline** [reply]
Bonjour Christophe, merci pour votre retour !

Le problème du buffer devrait être pris en charge via cette issue : https://github.com/isontheline/pro.webssh.net/issues/150

Le chantier devrait démarrer dans les prochaines semaines.

**Reply by isontheline** [reply]
@photochap Bonjour Christophe 👋

Avez-vous pu essayer les dernières versions de WebSSH ?
Améliorent-elles les points soulevés ?

**Comment by isontheline** [comment]
WebSSH 17.2 should fix these issues @photochap 

If it's not the case, feel free to come back to me
